### ISSUE TITLE
Invalid Memory Access after downloading two libraries through Pkg.add and using some classes

### ISSUE THREAD

**Issue by Bradan** [JuliaLang/julia#10913]
Hello,

I don't think this is very well reproducible, but simply copying and pasting the following lines into the julia environment made julia crash on arch linux:

    Pkg.add("Images");
    Pkg.add("ImageView");
    using Images, Color, FixedPointNumbers, ImageView;

    println("test")

The output:

    $ julia
                   _
       _       _ _(_)_     |  A fresh approach to technical computing
      (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
       _ _   _| |_  __ _   |  Type "help()" for help.
      | | | | | | |/ _` |  |
      | | |_| | | | (_| |  |  Version 0.3.7 (2015-03-23 21:36 UTC)
     _/ |\__'_|_|_|\__'_|  |  
    |__/                   |  x86_64-unknown-linux-gnu

    julia> Pkg.add("Images");
    INFO: Initializing package repository /home/myuser/.julia/v0.3
    INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
    INFO: Cloning cache of BinDeps from git://github.com/JuliaLang/BinDeps.jl.git
    INFO: Cloning cache of Color from git://github.com/JuliaLang/Color.jl.git
    INFO: Cloning cache of Compat from git://github.com/JuliaLang/Compat.jl.git
    INFO: Cloning cache of FixedPointNumbers from git://github.com/JeffBezanson/FixedPointNumbers.jl.git
    INFO: Cloning cache of Graphics from git://github.com/JuliaLang/Graphics.jl.git
    INFO: Cloning cache of Images from git://github.com/timholy/Images.jl.git
    INFO: Cloning cache of SHA from git://github.com/staticfloat/SHA.jl.git
    INFO: Cloning cache of SIUnits from git://github.com/Keno/SIUnits.jl.git
    INFO: Cloning cache of TexExtensions from git://github.com/Keno/TexExtensions.jl.git
    INFO: Cloning cache of URIParser from git://github.com/JuliaWeb/URIParser.jl.git
    INFO: Cloning cache of Zlib from git://github.com/dcjones/Zlib.jl.git
    INFO: Installing BinDeps v0.3.11
    INFO: Installing Color v0.4.4
    INFO: Installing Compat v0.4.1
    INFO: Installing FixedPointNumbers v0.0.7
    INFO: Installing Graphics v0.1.0
    INFO: Installing Images v0.4.36
    INFO: Installing SHA v0.0.4
    INFO: Installing SIUnits v0.0.2
    INFO: Installing TexExtensions v0.0.2
    INFO: Installing URIParser v0.0.5
    INFO: Installing Zlib v0.1.8
    INFO: Building Images
    INFO: Package database updated

    julia> Pkg.add("ImageView");
    INFO: Cloning cache of Cairo from git://github.com/JuliaLang/Cairo.jl.git
    INFO: Cloning cache of ImageView from git://github.com/timholy/ImageView.jl.git
    INFO: Cloning cache of IniFile from git://github.com/JuliaLang/IniFile.jl.git
    INFO: Cloning cache of Tk from git://github.com/JuliaLang/Tk.jl.git
    INFO: Cloning cache of Winston from git://github.com/nolta/Winston.jl.git
    INFO: Installing Cairo v0.2.26
    INFO: Installing ImageView v0.1.13
    INFO: Installing IniFile v0.2.4
    INFO: Installing Tk v0.3.1
    INFO: Installing Winston v0.11.9
    INFO: Building Cairo
    INFO: Building Images
    INFO: Building Tk
    INFO: Package database updated

    julia> using Images, Color, FixedPointNumbers, ImageView;

    signal (11): Speicherzugriffsfehler
    unknown function (ip: 457927034)
    unknown function (ip: 457927139)
    jl_get_binding at /usr/bin/../lib/julia/libjulia.so (unknown line)
    jl_get_global at /usr/bin/../lib/julia/libjulia.so (unknown line)
    jl_module_run_initializer at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457810210)
    unknown function (ip: 457808997)
    unknown function (ip: 457810800)
    jl_load at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include_from_node1 at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457742888)
    unknown function (ip: 457739417)
    unknown function (ip: 457808093)
    jl_f_top_eval at /usr/bin/../lib/julia/libjulia.so (unknown line)
    reload_path at loading.jl:152
    _require at loading.jl:67
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    require at loading.jl:54
    jlcall_require_20676 at  (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457803902)
    unknown function (ip: 457808683)
    unknown function (ip: 457808231)
    unknown function (ip: 457809809)
    unknown function (ip: 457808997)
    unknown function (ip: 457810800)
    jl_load at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include_from_node1 at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457742888)
    unknown function (ip: 457739417)
    unknown function (ip: 457808093)
    jl_f_top_eval at /usr/bin/../lib/julia/libjulia.so (unknown line)
    reload_path at loading.jl:152
    _require at loading.jl:67
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    require at loading.jl:54
    jlcall_require_20676 at  (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457803902)
    unknown function (ip: 457808683)
    unknown function (ip: 457808231)
    unknown function (ip: 457809809)
    unknown function (ip: 457808997)
    unknown function (ip: 457810800)
    jl_load at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    include_from_node1 at /usr/bin/../lib/julia/sys.so (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457742888)
    unknown function (ip: 457739417)
    unknown function (ip: 457808093)
    jl_f_top_eval at /usr/bin/../lib/julia/libjulia.so (unknown line)
    reload_path at loading.jl:152
    _require at loading.jl:67
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    require at loading.jl:51
    jlcall_require_20676 at  (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 457803902)
    unknown function (ip: 457808683)
    unknown function (ip: 457808231)
    jl_f_top_eval at /usr/bin/../lib/julia/libjulia.so (unknown line)
    eval_user_input at REPL.jl:53
    jlcall_eval_user_input_20063 at  (unknown line)
    jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
    anonymous at task.jl:95
    jl_handle_stack_switch at /usr/bin/../lib/julia/libjulia.so (unknown line)
    julia_trampoline at /usr/bin/../lib/julia/libjulia.so (unknown line)
    unknown function (ip: 4199613)
    __libc_start_main at /usr/bin/../lib/libc.so.6 (unknown line)
    unknown function (ip: 4199673)
    unknown function (ip: 0)
    Speicherzugriffsfehler (Speicherabzug geschrieben)

**Comment by timholy** [comment]
Does it crash with just the line starting at `using...`? Or only if `Pkg.add` is used?

**Comment by Bradan** [comment]
Only if Pkg.add is used and it downloads all the libraries. I couldn't reproduce it the second time when the libraries were already downloaded, but I thought the stacktrace might still be helpful for you.

**Comment by tkelman** [comment]
Little bit like https://github.com/JuliaLang/BinDeps.jl/issues/128?

**Comment by rened** [comment]
I hit the same tonight with `0.3.7` on both OSX and Ubuntu, running:
```jl
[try Pkg.add(x) end for x in Pkg.available()]
```
At least on Linux the error occurs only after the output of the comprehension `nothing ... nothing` is printed in the REPL, but before I can do anything else, so no `using ...` yet. 

My Linux error is the same as above, but perhaps the OSX version has a more informative stack trace:
```jl
signal (11): Segmentation fault: 11
ptrhash_get at /Users/rene/local/julia/src/support/ptrhash.c:24
jl_get_binding_ at /Users/rene/local/julia/src/module.c:144
jl_get_binding at /Users/rene/local/julia/src/module.c:332
jl_eval_module_expr at /Users/rene/local/julia/src/toplevel.c:180
jl_toplevel_eval_flex at /Users/rene/local/julia/src/toplevel.c:388
jl_parse_eval_all at /Users/rene/local/julia/src/toplevel.c:544
jl_load at /Users/rene/local/julia/src/toplevel.c:581
anonymous at ./no file:288
jl_apply at /Users/rene/local/julia/src/./julia.h:982
print_joined at ./string.jl:1422
jl_apply at /Users/rene/local/julia/src/./julia.h:982
jl_apply at /Users/rene/local/julia/src/interpreter.c:59
eval at /Users/rene/local/julia/src/interpreter.c:207
jl_toplevel_eval_flex at /Users/rene/local/julia/src/toplevel.c:496
jl_f_top_eval at /Users/rene/local/julia/src/builtins.c:399
_setindex! at ./dict.jl:538
rehash at ./dict.jl:419
jl_apply at /Users/rene/local/julia/src/./julia.h:982
cat at ./abstractarray.jl:639
cat at ./abstractarray.jl:656
jl_apply at /Users/rene/local/julia/src/./julia.h:982
anonymous at ./no file:0
jl_apply at /Users/rene/local/julia/src/./julia.h:982
jl_apply at /Users/rene/local/julia/src/toplevel.c:332
eval_import_path at /Users/rene/local/julia/src/toplevel.c:358
jl_parse_eval_all at /Users/rene/local/julia/src/toplevel.c:544
jl_load at /Users/rene/local/julia/src/toplevel.c:581
anonymous at ./no file:288
jl_apply at /Users/rene/local/julia/src/./julia.h:982
print_joined at ./string.jl:1422
jl_apply at /Users/rene/local/julia/src/./julia.h:982
jl_apply at /Users/rene/local/julia/src/interpreter.c:59
eval at /Users/rene/local/julia/src/interpreter.c:207
jl_toplevel_eval_flex at /Users/rene/local/julia/src/toplevel.c:496
jl_toplevel_eval_flex at /Users/rene/local/julia/src/toplevel.c:443
jl_f_top_eval at /Users/rene/local/julia/src/builtins.c:399
evalfile at loading.jl:168
jl_apply at /Users/rene/local/julia/src/./julia.h:982
anonymous at pkg/entry.jl:622
Dict at ./dict.jl:319
jl_apply at /Users/rene/local/julia/src/./julia.h:982
build! at pkg/entry.jl:621
build at pkg/entry.jl:633
jl_apply at /Users/rene/local/julia/src/./julia.h:982
_deepcopy_t at ./deepcopy.jl:0
edit at pkg/entry.jl:24
jl_apply at /Users/rene/local/julia/src/./julia.h:982
anonymous at task.jl:340
jl_apply at /Users/rene/local/julia/src/task.c:423
julia_trampoline at /Users/rene/local/julia/src/init.c:1032
fish: Job 1, 'julia' terminated by signal SIGSEGV (Address boundary error)
```
Installation method in both cases is `make` inside the top-level dir `~/local/julia`, but without `make install`. (this works fine for everything else I do)

**Comment by rened** [comment]
I just ran 
```jl
[try Pkg.add(x) end for x in Pkg.available()]
```
without problems on 0.3.11 and current master. I think this can be closed.

**Comment by tkelman** [comment]
Alright. If anyone can get this or similar segfaults to happen repeatably (or even often), either leave another comment here or open a new issue.
